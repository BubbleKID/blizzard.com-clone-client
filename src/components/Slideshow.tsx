import backgroundImage from '../images/wow-bcc-overlords-banner-desktop-2500x514.jpeg';

export default function Slideshow() {
    return (
        <div style={{height: '538px', backgroundImage: `url(${backgroundImage})`}}>
            im slide show
        </div>
    )
}