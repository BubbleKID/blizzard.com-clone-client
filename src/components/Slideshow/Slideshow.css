.slide-container {
    display: flex;
    align-items: center;
    bottom: 0;
    height: 514px;
    background-color: rgb(30, 3, 8);
    background-position: center top;
    background-size: auto;
    background-repeat: no-repeat;
    background-position: center top;
}

.slide-content {
    width: 100%;
    white-space: pre-wrap;
    padding: 56px;
    position: absolute;
    right: 0;
    left: 0;
}

.slide-title {
    font-size: 4.25rem;
    color: #fff;
    text-transform: uppercase;
    text-shadow: 4px 7px 17px rgb(0 0 0 / 80%);
    font-weight: 700;
    line-height: 1.2;
}

.slide-icon {
    display: block;
    margin-bottom: 24px;
}

.slide-btn {
    margin-top: 24px;
    background-color: #006fb2;
    border: solid 1px #00aeff;
    display: block;
    color: #fff;
    font-size: 18px;
    font-weight: 400;
    padding: 10px 50px;
    text-decoration: none;
    text-align: center;
    cursor: pointer;
    outline-offset: 2px;
}

.slide-btn:hover {
    background-color: #007bc6;
    border: solid 1px #00aeff;
}

.slider {
    background-color: #0b0d12;
    height: 24px;
    display: flex;
    align-items: center;
    justify-content: center;
}

.slider-item {
    width: 100px;
    height: 6px;
    margin: 0 4px;
    flex-grow: 0;
    background: rgba(255,255,255,.5);
    cursor: pointer;
}

.active {
    background: #0086ca !important;
    transform: scaleY(1.75);
}

.slide-control-btn {
    cursor: pointer;
    border: 1px solid #263945;
    background-color: rgba(0,0,0,.5);
    height: 80px;
    width: 40px;
    fill:  #00aeff;
    position: absolute;
}

.prev {
    left: 50%;
    right: auto;
    transform: translateX(-800px);
}

.next {
    left: auto;
    right: 50%;
    transform: translateX(800px);
}