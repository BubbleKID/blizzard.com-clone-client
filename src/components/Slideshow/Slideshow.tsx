import React, { useState, useEffect, useRef } from 'react';
import './Slideshow.css';
import  { ReactComponent as PreBtn } from '../../images/Blizzard Entertainment-10.svg';
import  { ReactComponent as NexBtn } from '../../images/Blizzard Entertainment-11.svg';

interface Props {
    slides: Slide[]
}

interface Slide {
    icon: string,
    iconWidth: number,
    title: string,
    btnTest: string,
    backgroundImage: string,
    backgroundColor: string
}

export default function Slideshow(props: Props) {
    const [running, setRunning] = useState(false);
    let [seconds, setSeconds] = useState<number>(3);
    const [slideControlButtonOpacity, setSlideControlButtonOpacity] = useState<number>(0);
    let [currentSlideIndex, setCurrentSlideIndex] = useState<number>(0);
    const btnRef = useRef<HTMLButtonElement>(null);
    const slideRef = useRef<HTMLDivElement>(null);
    const logoRef = useRef<HTMLImageElement>(null);
    const titleRef = useRef<HTMLSpanElement>(null);

    const onClickSlideControlBtn = (type: string): void => {
        let newIndex;

        if(type === 'prev') {
            startNewAnimation(slideRef, 'slide-shift-right');
            startNewAnimation(btnRef, 'shift-right-fade-in');
            startNewAnimation(logoRef, 'shift-right-fade-in');
            startNewAnimation(titleRef, 'shift-right-fade-in');
            if(currentSlideIndex === 0) {
                newIndex = props.slides.length - 1;
            } else {
                newIndex = currentSlideIndex - 1;
            }
        } else {
            startNewAnimation(slideRef, 'slide-shift-left');
            startNewAnimation(btnRef, 'shift-left-fade-in');
            startNewAnimation(logoRef, 'shift-left-fade-in');
            startNewAnimation(titleRef, 'shift-left-fade-in');
            if(currentSlideIndex === props.slides.length - 1) {
                newIndex = 0;
            } else {
                newIndex = currentSlideIndex + 1;
            }
        }
        setCurrentSlideIndex(newIndex);
    }

    const startNewAnimation = (ref: any, newClass: string) => {
        ref.current?.classList.forEach((className: string) => {
            if(className.includes('shift')) {
                ref.current?.classList.remove(className);
            }
        })
        ref.current?.offsetWidth;
        ref.current?.classList.add(newClass);
    }
    let isPlay = false;

    // useEffect(
    //     () => {
    //       if (running) {
    //         const interval = setInterval(() => {
    //             console.log(seconds)
    //             if(seconds > 0) {
    //                 setSeconds(seconds -= 1);
    //             } else {
    //               // setCurrentSlideIndex(currentSlideIndex += 1);
    //               clearInterval(interval);
    //               setSeconds(3);
    //               isPlay = true
    //             }
    //         }, 1000);
    //         return () => {
    //             setSeconds(3);
    //             clearInterval(interval);
    //             isPlay = false
    //         }
    //       }
    //     },
    //     [running]
    // );
    
    return (
        <React.Fragment>
            <div 
                className='slide-container'
                // onMouseEnter={() => {setSlideControlButtonOpacity(1); setRunning(false) }}
                // onMouseLeave={() => {setSlideControlButtonOpacity(0); setRunning(true) }}
                style={{backgroundColor: props.slides[currentSlideIndex].backgroundColor}}
            >
                <div ref={slideRef} className='slide-background slide-shift-left' style={{ backgroundImage: `url(${props.slides[currentSlideIndex].backgroundImage})`, backgroundColor: props.slides[currentSlideIndex].backgroundColor}}>
                    <div className="gradient-container"></div>
                </div>
                    <div className='slide-content'>                
                        <img ref={logoRef} className='slide-icon shift-left-fade-in' width={props.slides[currentSlideIndex] .iconWidth} src={props.slides[currentSlideIndex].icon}/>
                        <span ref={titleRef} className='slide-title shift-left-fade-in'>{props.slides[currentSlideIndex].title}</span>
                        <button ref={btnRef} className='slide-btn shift-left-fade-in' type='button'>{props.slides[currentSlideIndex].btnTest}</button>
                    </div>
                    <button className="slide-control-btn prev" style={{opacity: slideControlButtonOpacity}} onClick={() => onClickSlideControlBtn('prev')}><PreBtn/></button>
                    <button className="slide-control-btn next" style={{opacity: slideControlButtonOpacity}} onClick={() => onClickSlideControlBtn('next')}><NexBtn/></button>
            </div>
            <div className='slider'>
                { 
                    props.slides.map((slide, index) => 
                        <div key={index}
                            onClick={() => setCurrentSlideIndex(index)}
                            className={index === currentSlideIndex ? 'slider-item active':'slider-item'}
                        >
                            { index === currentSlideIndex && isPlay ? ( <div className='play'></div>) : null }
                        </div>
                    )
                }
            </div>
        </React.Fragment>
    )
}