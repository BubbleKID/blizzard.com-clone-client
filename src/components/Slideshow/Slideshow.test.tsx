import React from 'react';
import TestRenderer  from 'react-test-renderer';
import Slideshow from './Slideshow';
import tbcIcon from '../../images/icons/WOW_Classic_Phase_2p5p2_Logo_01.png';
import d2Icon from '../../images/icons/D2R_DiabloResurrected_Logo_blizz.png';
import backgroundImageTbc from '../../images/wow-bcc-overlords-banner-desktop-2500x514.jpeg';
import backgroundImageD2 from '../../images/D2R_2021_Blizzard.comDesktop_2500x514_MB01.png';

test('Slideshow should be rendered', () => {
    const slides = [
        {
            icon: tbcIcon,
            iconWidth: 260,
            btnTest: 'Play Today',
            title: 'Illidan\'s lieutenants\nAwait',
            backgroundImage: backgroundImageTbc,
            backgroundColor: 'rgb(30, 3, 8)'
        },
        {
            icon: d2Icon,
            iconWidth: 260,
            btnTest: 'Buy Now',
            title: 'Evil Is Resurrected',
            backgroundImage: backgroundImageD2,
            backgroundColor: 'rgb(0, 0, 0)'
        }
    ];
    const component = TestRenderer.create(<Slideshow slides={slides}/>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
