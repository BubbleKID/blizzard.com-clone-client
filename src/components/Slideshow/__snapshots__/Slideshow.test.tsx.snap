// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slideshow should be rendered 1`] = `
Array [
  <div
    className="slide-container"
    style={
      Object {
        "backgroundColor": "rgb(30, 3, 8)",
      }
    }
  >
    <div
      className="slide-background slide-shift-left"
      style={
        Object {
          "backgroundColor": "rgb(30, 3, 8)",
          "backgroundImage": "url()",
        }
      }
    >
      <div
        className="gradient-container"
      />
    </div>
    <div
      className="slide-content"
    >
      <img
        className="slide-icon shift-left-fade-in"
        src=""
        width={260}
      />
      <span
        className="slide-title shift-left-fade-in"
      >
        Illidan's lieutenants
Await
      </span>
      <button
        className="slide-btn shift-left-fade-in"
        type="button"
      >
        Play Today
      </button>
    </div>
    <button
      className="slide-control-btn prev"
      onClick={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <span />
    </button>
    <button
      className="slide-control-btn next"
      onClick={[Function]}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <span />
    </button>
  </div>,
  <div
    className="slider"
  >
    <div
      className="slider-item active"
      onClick={[Function]}
    />
    <div
      className="slider-item"
      onClick={[Function]}
    />
    <div
      className="slider-item"
      onClick={[Function]}
    />
  </div>,
]
`;
